client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ExecuteWatchRequest.java

```diff
@@ -0,0 +1,167 @@
+     * @param watchContent the JSON definition of the watch
+     */
+    public static ExecuteWatchRequest inline(String watchContent) {
+        return new ExecuteWatchRequest(null, Objects.requireNonNull(watchContent, "Watch content cannot be null"));
+    }
+
+    private ExecuteWatchRequest(String id, String watchContent) {
+        this.id = id;
+        this.watchContent = watchContent == null ? null : new BytesArray(watchContent);
+    }
+
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * @param ignoreCondition set if the condition for this execution be ignored
+     */
+    public void setIgnoreCondition(boolean ignoreCondition) {
+        this.ignoreCondition = ignoreCondition;
+    }
+
+    public boolean ignoreCondition() {
+        return ignoreCondition;
+    }
+
+    /**
+     * @param recordExecution Sets if this execution be recorded in the history index
+     */
+    public void setRecordExecution(boolean recordExecution) {
+        if (watchContent != null && recordExecution) {
+            throw new IllegalArgumentException("The execution of an inline watch cannot be recorded");
+        }
+        this.recordExecution = recordExecution;
+    }
+
+    public boolean recordExecution() {
+        return recordExecution;
+    }
+
+    /**
+     * @param alternativeInput Sets the alternative input
+     */
+    public void setAlternativeInput(String alternativeInput) {
+        this.alternativeInput = new BytesArray(alternativeInput);
+    }
+
+    /**
+     * @param data A JSON string representing the data that should be associated with the trigger event.
+     */
+    public void setTriggerData(String data) {
+        this.triggerData = new BytesArray(data);
+    }
+
+    /**
+     * Sets the action execution mode for the give action (identified by its id).
+     *
+     * @param actionId      the action id.
+     * @param actionMode    the execution mode of the action.
+     */
+    public void setActionMode(String actionId, ActionExecutionMode actionMode) {
+        Objects.requireNonNull(actionId, "actionId cannot be null");
+        actionModes.put(actionId, actionMode);
+    }
+
+    public Map<String, ActionExecutionMode> getActionModes() {
+        return this.actionModes;
+    }
+
+    /**
+     * @param debug indicates whether the watch should execute in debug mode. In debug mode the
+     *              returned watch record will hold the execution {@code vars}
+     */
+    public void setDebug(boolean debug) {
+        this.debug = debug;
+    }
+
+    public boolean isDebug() {
+        return debug;
+    }
+
+    @Override
+    public String toString() {
+        return "execute[" + id + "]";
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (triggerData != null) {
+            builder.rawField("trigger_data", triggerData.streamInput(), XContentType.JSON);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *streamInput*. Currently, there are execution paths that do not contain closure statements, for example, when XContentBuilder.rawField throws an exception. Ensure the object returned by *streamInput()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).